arm-batch/2017-01-01/swagger/BatchManagement.json

```diff
@@ -0,0 +1,1563 @@
+      "properties": {
+        "autoStorage": {
+          "$ref": "#/definitions/AutoStorageBaseProperties",
+          "description": "The properties related to auto storage account."
+        }
+      },
+      "description": "The properties of a Batch account."
+    },
+    "BatchAccountCreateParameters": {
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "The region in which to create the account."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The user specified tags associated with the account."
+        },
+        "poolAllocationMode": {
+          "type": "string",
+          "title": "The allocation mode to use for creating pools in the Batch account.",
+          "description": "The pool allocation mode also affects how clients may authenticate to the Batch Service API. If the mode is BatchService, clients may authenticate using access keys or Azure Active Directory. If the mode is UserSubscription, clients must use Azure Active Directory. The default is BatchService.",
+          "enum": [
+            "BatchService",
+            "UserSubscription"
+          ],
+          "x-ms-enum": {
+            "name": "PoolAllocationMode",
+            "modelAsString": false
+          }
+        },
+        "keyVaultReference": {
+          "$ref": "#/definitions/KeyVaultReference",
+          "description": "A reference to the Azure key vault associated with the Batch account."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/BatchAccountBaseProperties",
+          "description": "The properties of the Batch account."
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "description": "Parameters supplied to the Create operation."
+    },
+    "KeyVaultReference": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The resource ID of the Azure key vault associated with the Batch account."
+        }
+      },
+      "required": [
+        "id"
+      ],
+      "description": "Identifies the Azure key vault associated with a Batch account."
+    },
+    "AutoStorageProperties": {
+      "properties": {
+        "storageAccountId": {
+          "type": "string",
+          "description": "The resource ID of the storage account to be used for auto storage account."
+        },
+        "lastKeySync": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The UTC time at which storage keys were last synchronized with the Batch account."
+        }
+      },
+      "required": [
+        "storageAccountId",
+        "lastKeySync"
+      ],
+      "description": "Contains information about the auto storage account associated with a Batch account."
+    },
+    "BatchAccountProperties": {
```

Comment:
There is `"x-nullable": false` if it is about the user experience. Since this is a response-only model (correct?), it would be somewhat weird to make some properties readonly while others are not...